File: src/main/kotlin/com/ding/miniminimoney/owe/api/LoanRestController.java
package com.ding.miniminimoney.owe.api;


import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(
        "/loan"
)
class LoanRestController(
        private val oweService: OweService
) {

    @PostMapping("/create/{userId}")
    fun createWallet(
            @PathVariable("userId") userId: String
    ): ResponseEntity<WalletResponseBody> {
        val walletCreated = walletService.createAndAssignWallet(userId)
        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(walletCreated.toWalletResponseBody())
    }
}

File: src/main/kotlin/com/ding/miniminimoney/transaction/entities/api/LoanRestController.kt
package com.ding.miniminimoney.transaction.entities.api;


import com.ding.miniminimoney.owe.rs.dto.LoanRequestBody
import com.ding.miniminimoney.owe.rs.dto.LoanResponseBody
import com.ding.miniminimoney.owe.services.LoanService
import org.springframework.http.HttpMethod
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(
        "/loan"
)
class LoanRestController(
        private val loanService: LoanService
) {

    @PostMapping("/createLoan/{userId}")
    fun createLoan(
        @PathVariable("userId") userId: String,

        @RequestBody
        createLoanRequestBody: LoanRequestBody

    ): ResponseEntity<LoanResponseBody> {
        val loanCreated = loanService.createLoan(userId, createLoanRequestBody)
        return ResponseEntity
                .status(HttpStatus.CREATED)
                .allow(HttpMethod.GET)
                .body(loanCreated)
    }
}
